node/test/util/push.js

```diff
@@ -72,19 +224,22 @@ describe("push", function () {
             initial: "a=B|b=B|x=Ca:I b=Sb:1;Ob Bmaster=1",
             manipulator: pusher("x", "origin", "master", "master"),
         },
-        "open submodule make a branch": {
-            initial: "a=B|b=B|x=Ca:I b=Sb:1;Ob Bmaster=1",
+        "open submodule make an unneeded ref": {
+            initial: "a=B|b=B|x=Ca:C2-1 b=Sb:1;Bmaster=2;Ob",
             manipulator: pusher("x", "origin", "master", "foo"),
             expected: "\
-a=E:Bfoo=1|\
-b=E:Bfoo=1|\
-x=E:Rorigin=a master=1,foo=1;Ob Bmaster=1!Rorigin=b master=1,foo=1",
+a=E:Bfoo=2|\
+b=E:Fcommits/1=1|\
+x=E:Rorigin=a master=1,foo=2;Ob Fcommits/1=1",
         },
-        "open submodule fails, no meta update": {
-            initial: "a=B|b=B|x=Ca:I b=Sb:1;Ob",
+        "open submodule push new ref": {
```

Comment:
This was the case where we failed to push a submodule because it didn't have the branch (`foo` in this case) that we were trying to push; that case is no longer applicable.